packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -473,14 +685,17 @@ class RenderParagraph extends RenderBox {
     _recognizerOffsets.clear();
     _recognizers.clear();
     int offset = 0;
-    text.visitTextSpan((TextSpan span) {
-      if (span.recognizer != null && (span.recognizer is TapGestureRecognizer || span.recognizer is LongPressGestureRecognizer)) {
-        final int length = span.semanticsLabel?.length ?? span.text.length;
+    text.visitInlineSpan((InlineSpan span) {
+      TextSpan textSpan = InlineSpan.asType<TextSpan>(span);
+      if (textSpan == null)
+        return true;
```

Comment:
this seems dubious. it looks like the semantics will end up all wrong.

Please add a test that verifies that if the paragraph has a text span "A" followed by a widget with semantics "B" followed by a text span "C", the semantics tree ends up saying "A", "B", "C".